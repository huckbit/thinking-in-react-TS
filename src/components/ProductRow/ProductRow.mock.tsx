import type { Product } from "./ProductRow";

const mock: Product = {
  category: "Fruits",
  price: "Â£1",
  stocked: true,
  name: "Apple",
};

export default mock;
