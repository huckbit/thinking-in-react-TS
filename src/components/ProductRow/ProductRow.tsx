export interface Product {
  category: string;
  price: string;
  stocked: boolean;
  name: string;
}

function ProductRow({ product }: { product: Product }) {
  const { stocked, name, price } = product;
  const productName = stocked ? name : <span style={{ color: "red" }}>{name}</span>;
  return (
    <tr>
      <td>{productName}</td>
      <td>{price}</td>
    </tr>
  );
}

export default ProductRow;
