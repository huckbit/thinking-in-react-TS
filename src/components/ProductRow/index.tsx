export { default } from "./ProductRow";
export type { Product } from "./ProductRow";
