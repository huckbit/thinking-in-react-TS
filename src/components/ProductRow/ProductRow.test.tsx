/// <reference types="@testing-library/jest-dom" />
import { render, screen } from "@testing-library/react";
import ProductRow from "./ProductRow";
import mock from "./ProductRow.mock";

describe("ProductRow", () => {
  test("it matches the snapshot", () => {
    const { asFragment } = render(
      <table>
        <tbody>
          <ProductRow product={mock} />
        </tbody>
      </table>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders product name and price", () => {
    render(
      <table>
        <tbody>
          <ProductRow product={mock} />
        </tbody>
      </table>,
    );

    expect(screen.getByText(mock.name)).toBeInTheDocument();
    expect(screen.getByText(mock.price)).toBeInTheDocument();
  });
});
