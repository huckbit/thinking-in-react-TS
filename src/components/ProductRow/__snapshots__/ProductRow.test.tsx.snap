// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRow it matches the snapshot 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr>
        <td>
          Apple
        </td>
        <td>
          Â£1
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
